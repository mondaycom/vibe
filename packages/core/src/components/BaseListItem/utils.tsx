import { EndElement, StartElement } from "./BaseListItem.types";
import { TextType } from "../Text";
import React from "react";
import Avatar from "../Avatar/Avatar";
import styles from "./BaseListItem.module.scss";
import Icon from "../Icon/Icon";
import Text from "../Text/Text";

export function renderSideElement(
  element: StartElement | EndElement,
  disabled: boolean,
  textVariant: TextType
): React.ReactNode {
  switch (element.type) {
    case "avatar":
      return (
        <Avatar
          className={styles.avatar}
          src={element.value}
          withoutBorder
          square={element.square}
          type="img"
          customSize={20}
          disabled={disabled}
        />
      );

    case "icon":
      return <Icon className={styles.icon} icon={element.value} iconSize={20} />;

    case "indent":
      return <div className={styles.indent} />;

    case "suffix":
      return (
        <Text type={textVariant} color="secondary" style={{ maxWidth: "112px" }}>
          {element.value}
        </Text>
      );

    case "custom":
      return element.render();
  }
}
