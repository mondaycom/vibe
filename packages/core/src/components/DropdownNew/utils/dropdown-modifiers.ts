import { Modifier } from "react-popper";

export const matchWidthModifier: Modifier<any>[] = [
  {
    name: "matchWidth",
    enabled: true,
    phase: "beforeWrite",
    requires: ["computeStyles"],
    fn: ({ state }) => {
      if (state.rects && state.rects.reference) {
        state.styles.popper.width = `${state.rects.reference.width}px`;
      }
    }
  }
];
