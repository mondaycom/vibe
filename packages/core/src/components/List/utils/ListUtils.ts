import { MutableRefObject, useState } from "react";
import { ListItemElement } from "../../ListItem/ListItem.types";
import { ListElement } from "../List.types";
import useIsomorphicLayoutEffect from "../../../hooks/ssr/useIsomorphicLayoutEffect";

let listIdCounter = 0;
export const generateListId = () => {
  return `list-${listIdCounter++}`;
};

const VALID_ROLES = ["option", "listitem", "menuitem", "tab", "treeitem"];

export const useListId = (id: string) => {
  const [listId, setListId] = useState<string>();
  useIsomorphicLayoutEffect(() => {
    setListId(id || generateListId());
  }, [id]);
  return listId;
};

export const getListItemIdByIndex = (childrenRefs: MutableRefObject<HTMLElement[]>, index: number): string => {
  return childrenRefs.current[index]?.id;
};

export const getListItemIndexById = (childrenRefs: MutableRefObject<HTMLElement[]>, id: string): number => {
  return childrenRefs.current.findIndex(child => child?.id === id);
};

export const getListItemComponentType = (listComponent: ListElement): ListItemElement => {
  switch (listComponent) {
    case "ul":
    case "ol":
      return "li";
    case "nav":
      return "a";
    default:
      return "div";
  }
};

export const isListItem = (element: HTMLElement) => {
  return element && element instanceof HTMLElement && VALID_ROLES.includes(element.getAttribute("role"));
};

export const getNextListItemIndex = (currentIndex: number, childrenRefs: MutableRefObject<HTMLElement[]>) => {
  while (currentIndex < childrenRefs.current.length - 1) {
    const child = childrenRefs.current[++currentIndex];
    if (isListItem(child)) {
      return currentIndex;
    }
  }
  return undefined;
};

export const getPrevListItemIndex = (currentIndex: number, childrenRefs: MutableRefObject<HTMLElement[]>) => {
  while (currentIndex > 0) {
    const child = childrenRefs.current[--currentIndex];
    if (isListItem(child)) {
      return currentIndex;
    }
  }
  return undefined;
};
