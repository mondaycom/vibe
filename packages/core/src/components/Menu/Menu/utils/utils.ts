import { type ReactElement } from "react";
import { type MenuChild } from "../MenuConstants";

export function isMenuChildSelectable(child: MenuChild): boolean {
  return !!child.type.isSelectable && !child.props.disabled;
}

export const generateMenuItemId = (menuId: string, child: ReactElement, index: number) => {
  return (child.props as { id: string }).id || `${menuId}-item-${index}`;
};
