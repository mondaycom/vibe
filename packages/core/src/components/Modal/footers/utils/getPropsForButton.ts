import { ModalFooterActionProps } from "../ModalFooterBase/ModalFooterBase.types";
import cx from "classnames";

export function getPropsForButton(button?: ModalFooterActionProps, buttonClassName?: string) {
  if (!button) return undefined;
  const { tooltipProps, className, ...rest } = button;
  return {
    ...rest,
    className: tooltipProps?.content ? className : cx(className, buttonClassName),
    tooltipProps: tooltipProps?.content
      ? { ...tooltipProps, referenceWrapperClassName: cx(tooltipProps.referenceWrapperClassName, buttonClassName) }
      : undefined
  };
}
