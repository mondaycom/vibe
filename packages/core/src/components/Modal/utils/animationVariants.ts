import { Variants } from "framer-motion";
import { RefObject } from "react";

const enterEase = [0.0, 0.0, 0.4, 1.0];
const exitEase = [0.6, 0.0, 1.0, 1.0];

export const modalAnimationOverlayVariants: Variants = {
  initial: {
    opacity: 0
  },
  enter: {
    opacity: 1,
    transition: {
      duration: 0.1,
      ease: enterEase
    }
  },
  exit: {
    opacity: 0,
    transition: {
      duration: 0.1,
      ease: exitEase
    }
  }
};

export const modalAnimationCenterPopVariants: Variants = {
  initial: {
    opacity: 1,
    scale: 1,
    x: "-50%",
    y: "-50%"
  },
  enter: {
    opacity: [0, 1, 1],
    scale: [0.8, 1, 1],
    x: "-50%",
    y: "-50%",
    transition: {
      duration: 0.15,
      ease: enterEase,
      times: [0, 0.5, 1]
    }
  },
  exit: {
    opacity: [1, 1, 0],
    scale: [1, 1, 0.8],
    x: "-50%",
    y: "-50%",
    transition: {
      duration: 0.1,
      ease: exitEase,
      times: [0, 0.5, 1]
    }
  }
};

export const modalAnimationAnchorPopVariants: Variants = {
  initial: (anchorElementRef: RefObject<HTMLElement>) => {
    const anchorRect = anchorElementRef.current.getBoundingClientRect();
    const anchorCenterX = anchorRect.left + anchorRect.width / 2;
    const anchorCenterY = anchorRect.top + anchorRect.height / 2;

    const x = `calc(-50% + ${anchorCenterX}px)`;
    const y = `calc(-50% + ${anchorCenterY}px)`;

    return {
      opacity: 0,
      scale: 0.8,
      top: "50%",
      left: "50%",
      x,
      y
    };
  },
  enter: {
    opacity: [0, 0, 1],
    scale: [0.8, 0.8, 1],
    top: "50%",
    left: "50%",
    x: "-50%",
    y: "-50%",
    transition: {
      duration: 0.2,
      ease: enterEase,
      times: [0, 0.4, 1]
    }
  },
  exit: (anchorElementRef: RefObject<HTMLElement>) => {
    const anchorRect = anchorElementRef.current.getBoundingClientRect();
    const anchorCenterX = anchorRect.left + anchorRect.width / 2;
    const anchorCenterY = anchorRect.top + anchorRect.height / 2;

    const x = `calc(-50% + ${anchorCenterX}px)`;
    const y = `calc(-50% + ${anchorCenterY}px)`;

    return {
      opacity: [1, 0, 0],
      scale: [1, 0.8, 0.8],
      top: 0,
      left: 0,
      x,
      y,
      transition: {
        duration: 0.15,
        ease: exitEase,
        times: [0, 0.6, 1]
      }
    };
  }
};

export const modalAnimationFullViewVariants: Variants = {
  enter: {
    opacity: [0.3, 1, 1],
    y: [30, 0],
    transition: {
      duration: 0.25,
      ease: enterEase,
      times: [0, 0.33, 1]
    }
  },
  exit: {
    opacity: 0,
    y: 30,
    transition: {
      duration: 0.1,
      ease: exitEase
    }
  }
};
