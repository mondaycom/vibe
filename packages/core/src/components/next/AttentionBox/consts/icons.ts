import { Info, Check, Alert, Warning, Favorite } from "@vibe/icons";
import { AttentionBoxType } from "../AttentionBox.types";
import { SubIcon } from "../../../../types";

export const ATTENTION_BOX_DEFAULT_ICONS: Record<AttentionBoxType, SubIcon> = {
  primary: Info,
  success: Check,
  danger: Alert,
  warning: Warning,
  dark: Favorite
};
