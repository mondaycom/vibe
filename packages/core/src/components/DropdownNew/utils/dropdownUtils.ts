import { BaseListItemData } from "../../BaseListItem";
import { ListGroup } from "../../BaseList";

export function normalizeOptions<Item extends BaseListItemData>(
  options: ListGroup<Item>[] | BaseListItemData<Item>[],
  filter?: string,
  filterOption?: (option: Item, inputValue: string) => boolean
): ListGroup<Item>[] {
  let indexCounter = 0;
  const defaultFilterFn = (item: Item, inputValue: string) =>
    !inputValue || item.label.toLowerCase().includes(inputValue.toLowerCase());
  const currentFilterFn = filterOption || defaultFilterFn;

  return Array.isArray(options) && options.some(item => "options" in item)
    ? (options as ListGroup<Item>[]).flatMap(group => {
        const filteredOptions = group.options
          .filter(item => currentFilterFn(item, filter || ""))
          .map(item => ({ ...item, index: indexCounter++ }));

        return filteredOptions.length > 0 ? [{ ...group, options: filteredOptions }] : [];
      })
    : [
        {
          label: undefined,
          options: (options as BaseListItemData<Item>[])
            .filter(item => currentFilterFn(item, filter || ""))
            .map(item => ({ ...item, index: indexCounter++ }))
        }
      ];
}
