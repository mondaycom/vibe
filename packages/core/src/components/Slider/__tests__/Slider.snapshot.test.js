import React from "react";
import { cleanup } from "@testing-library/react";
import { snapshotDiff } from "../../../utils/jest-utils";
import { Sound } from "@vibe/icons";
import { renderSliderForSnapshots } from "./sliderTestUtils";

vi.mock("../../Icon/Icon", () => {
  return ({ icon, ...rest }) => {
    return <div data-testid="mock-icon">{JSON.stringify({ ...rest, icon: icon && icon.displayName })}</div>;
  };
});

vi.mock("../../TextField/TextField", () => {
  const TextField = props => {
    return <div data-testid="mock-text-field-comp">{JSON.stringify(props)}</div>;
  };
  return TextField;
});

describe("Slider Render", () => {
  let defaultRender;
  beforeAll(async () => {
    defaultRender = await renderSliderForSnapshots();
    cleanup();
  });

  afterEach(() => {
    cleanup();
  });

  it("01. with empty props", async () => {
    expect(defaultRender).toMatchSnapshot();
  });

  it("02. with ariaLabel", async () => {
    const props = { ariaLabel: "My Slider" };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("03. with ariaLabelledby", async () => {
    const props = { ariaLabelledby: "id-of-the-custom-label" };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("04. with custom className", async () => {
    const props = { className: "slide-me" };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("05. with id and data-testid", async () => {
    const props = { id: "my-slider", "data-testid": "my-slider" };
    const currentRender = await renderSliderForSnapshots(props, "my-slider__thumb-0");
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("06. with disabled", async () => {
    const props = { disabled: true };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it("07. with color", async () => {
    const props = { color: "positive" };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`08. with size`, async () => {
    const props = { size: "large" };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`09. with min, max, and step`, async () => {
    const props = { min: 10, max: 20, step: 2 };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`10. with showValue`, async () => {
    const props = { showValue: true };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`11. with valueFormatter`, async () => {
    const props = { valueFormatter: value => `--${value}GB--` };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`12. with value (controlled component)`, async () => {
    const props = { value: 12 };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`13. with value, valueText (controlled component)`, async () => {
    const props = { value: 15, valueText: "15kg" };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`14. with indicateSelection`, async () => {
    const props = { indicateSelection: true };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`15. with prefix (string) and postfix (Icon)`, async () => {
    const props = { prefix: "Vol", postfix: { icon: Sound } };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender)).toMatchSnapshot();
  });

  it(`16. with prefix (custom) and postfix (render-props)`, async () => {
    const props = {
      prefix: <div>Custom component</div>,
      postfix: (value, valueText) => `Render Props result: ${valueText} (${value})`
    };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender)).toMatchSnapshot();
  });

  it(`17. with ranged (ranged slider)`, async () => {
    const props = { ranged: true };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });

  it(`18. ranged slider with options`, async () => {
    const props = { defaultValue: [20, 30], min: 10, max: 50, ranged: true, step: 2 };
    const currentRender = await renderSliderForSnapshots(props);
    expect(snapshotDiff(defaultRender, currentRender, { props })).toMatchSnapshot();
  });
});
