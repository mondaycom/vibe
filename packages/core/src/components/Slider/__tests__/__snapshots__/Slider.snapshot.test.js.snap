// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Render 01. with empty props 1`] = `
<div
  class="slider positionTop"
  data-testid="monday-slider"
>
  <div
    class="base small"
    data-testid="monday-slider__base"
  >
    <div
      class="rail small"
      data-testid="monday-slider__rail"
    >
      <div
        class="track"
      />
      <div
        class="filledTrack"
        style="left: 0%; width: 0%;"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        aria-valuetext="0%"
        class="thumb small notDisabledThumb"
        data-testid="monday-slider__thumb-0"
        role="slider"
        style="left: 0%;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;

exports[`Slider Render 02. with ariaLabel 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"ariaLabel":"My Slider"}

@@ --- --- @@
        />
        <div
          aria-disabled="false"
+         aria-label="My Slider"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
`;

exports[`Slider Render 03. with ariaLabelledby 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"ariaLabelledby":"id-of-the-custom-label"}

@@ --- --- @@
        />
        <div
          aria-disabled="false"
+         aria-labelledby="id-of-the-custom-label"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
`;

exports[`Slider Render 04. with custom className 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"className":"slide-me"}

@@ --- --- @@
  <div
-   class="slider positionTop"
+   class="slider positionTop slide-me"
    data-testid="monday-slider"
  >
    <div
`;

exports[`Slider Render 05. with id and data-testid 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"id":"my-slider","data-testid":"my-slider"}

@@ --- --- @@
  <div
    class="slider positionTop"
-   data-testid="monday-slider"
+   data-testid="my-slider"
+   id="my-slider"
  >
    <div
      class="base small"
-     data-testid="monday-slider__base"
+     data-testid="my-slider__base"
    >
      <div
        class="rail small"
-       data-testid="monday-slider__rail"
+       data-testid="my-slider__rail"
      >
        <div
          class="track"
@@ --- --- @@
          aria-valuenow="0"
          aria-valuetext="0%"
          class="thumb small notDisabledThumb"
-         data-testid="monday-slider__thumb-0"
+         data-testid="my-slider__thumb-0"
          role="slider"
          style="left: 0%;"
          tabindex="0"
`;

exports[`Slider Render 06. with disabled 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"disabled":true}

@@ --- --- @@
    data-testid="monday-slider"
  >
    <div
-     class="base small"
+     class="base small disabled"
      data-testid="monday-slider__base"
    >
      <div
@@ --- --- @@
          style="left: 0%; width: 0%;"
        />
        <div
-         aria-disabled="false"
+         aria-disabled="true"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
          aria-valuetext="0%"
-         class="thumb small notDisabledThumb"
+         class="thumb small"
          data-testid="monday-slider__thumb-0"
          role="slider"
          style="left: 0%;"
-         tabindex="0"
+         tabindex="-1"
        />
      </div>
    </div>
`;

exports[`Slider Render 07. with color 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"color":"positive"}

@@ --- --- @@
    data-testid="monday-slider"
  >
    <div
-     class="base small"
+     class="base small positive"
      data-testid="monday-slider__base"
    >
      <div
@@ --- --- @@
        data-testid="monday-slider__rail"
      >
        <div
-         class="track"
+         class="track positive"
        />
        <div
-         class="filledTrack"
+         class="filledTrack positive"
          style="left: 0%; width: 0%;"
        />
        <div
@@ --- --- @@
          aria-valuemin="0"
          aria-valuenow="0"
          aria-valuetext="0%"
-         class="thumb small notDisabledThumb"
+         class="thumb positive small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
          style="left: 0%;"
`;

exports[`Slider Render 08. with size 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"size":"large"}

@@ --- --- @@
    data-testid="monday-slider"
  >
    <div
-     class="base small"
+     class="base large"
      data-testid="monday-slider__base"
    >
      <div
-       class="rail small"
+       class="rail large"
        data-testid="monday-slider__rail"
      >
        <div
@@ --- --- @@
          aria-valuemin="0"
          aria-valuenow="0"
          aria-valuetext="0%"
-         class="thumb small notDisabledThumb"
+         class="thumb large notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
          style="left: 0%;"
`;

exports[`Slider Render 09. with min, max, and step 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"min":10,"max":20,"step":2}

@@ --- --- @@
        />
        <div
          aria-disabled="false"
-         aria-valuemax="100"
-         aria-valuemin="0"
-         aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuemax="20"
+         aria-valuemin="10"
+         aria-valuenow="10"
+         aria-valuetext="10%"
          class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
`;

exports[`Slider Render 10. with showValue 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"showValue":true}

@@ --- --- @@
  <div
-   class="slider positionTop"
+   class="slider valueShown positionTop"
    data-testid="monday-slider"
  >
    <div
@@ --- --- @@
          role="slider"
          style="left: 0%;"
          tabindex="0"
-       />
+       >
+         <label
+           class="label colorPrimary positionTop"
+         >
+           0%
+         </label>
+       </div>
      </div>
    </div>
  </div>
`;

exports[`Slider Render 11. with valueFormatter 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {}

@@ --- --- @@
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuetext="--0GB--"
          class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
`;

exports[`Slider Render 12. with value (controlled component) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"value":12}

@@ --- --- @@
        />
        <div
          class="filledTrack"
-         style="left: 0%; width: 0%;"
+         style="left: 0%; width: 12%;"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
-         aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuenow="12"
+         aria-valuetext="12%"
          class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
-         style="left: 0%;"
+         style="left: 12%;"
          tabindex="0"
        />
      </div>
`;

exports[`Slider Render 13. with value, valueText (controlled component) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"value":15,"valueText":"15kg"}

@@ --- --- @@
        />
        <div
          class="filledTrack"
-         style="left: 0%; width: 0%;"
+         style="left: 0%; width: 15%;"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
-         aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuenow="15"
+         aria-valuetext="15kg"
          class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
-         style="left: 0%;"
+         style="left: 15%;"
          tabindex="0"
        />
      </div>
`;

exports[`Slider Render 14. with indicateSelection 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"indicateSelection":true}

@@ --- --- @@
        />
      </div>
    </div>
+   <div
+     class="infix postfix"
+     style="width: 60px;"
+   >
+     <div
+       data-testid="mock-text-field-comp"
+     >
+       {"value":"0%","className":"selectionIndicator"}
+     </div>
+   </div>
  </div>
`;

exports[`Slider Render 15. with prefix (string) and postfix (Icon) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
    data-testid="monday-slider"
  >
    <div
+     class="infix prefix txt"
+   >
+     Vol
+   </div>
+   <div
      class="base small"
      data-testid="monday-slider__base"
    >
@@ --- --- @@
          style="left: 0%;"
          tabindex="0"
        />
+     </div>
+   </div>
+   <div
+     class="infix postfix"
+   >
+     <div
+       data-testid="mock-icon"
+     >
+       {"clickable":false,"iconSize":18,"ignoreFocusStyle":true,"icon":"Sound"}
      </div>
    </div>
  </div>
`;

exports[`Slider Render 16. with prefix (custom) and postfix (render-props) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render

@@ --- --- @@
    data-testid="monday-slider"
  >
    <div
+     class="infix prefix"
+   >
+     <div>
+       Custom component
+     </div>
+   </div>
+   <div
      class="base small"
      data-testid="monday-slider__base"
    >
@@ --- --- @@
          tabindex="0"
        />
      </div>
+   </div>
+   <div
+     class="infix postfix"
+   >
+     Render Props result: 0% (0)
    </div>
  </div>
`;

exports[`Slider Render 17. with ranged (ranged slider) 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"ranged":true}

@@ --- --- @@
        />
        <div
          class="filledTrack"
-         style="left: 0%; width: 0%;"
+         style="left: 0%; width: 100%;"
        />
        <div
          aria-disabled="false"
@@ --- --- @@
          data-testid="monday-slider__thumb-0"
          role="slider"
          style="left: 0%;"
+         tabindex="0"
+       />
+       <div
+         aria-disabled="false"
+         aria-valuemax="100"
+         aria-valuemin="0"
+         aria-valuenow="100"
+         aria-valuetext="100%"
+         class="thumb small notDisabledThumb"
+         data-testid="monday-slider__thumb-1"
+         role="slider"
+         style="left: 100%;"
          tabindex="0"
        />
      </div>
`;

exports[`Slider Render 18. ranged slider with options 1`] = `
Snapshot Diff:
- Default Render
+ Current Render: {"defaultValue":[20,30],"min":10,"max":50,"ranged":true,"step":2}

@@ --- --- @@
        />
        <div
          class="filledTrack"
-         style="left: 0%; width: 0%;"
+         style="left: 25%; width: 25%;"
        />
        <div
          aria-disabled="false"
-         aria-valuemax="100"
-         aria-valuemin="0"
-         aria-valuenow="0"
-         aria-valuetext="0%"
+         aria-valuemax="50"
+         aria-valuemin="10"
+         aria-valuenow="20"
+         aria-valuetext="20%"
          class="thumb small notDisabledThumb"
          data-testid="monday-slider__thumb-0"
          role="slider"
-         style="left: 0%;"
+         style="left: 25%;"
+         tabindex="0"
+       />
+       <div
+         aria-disabled="false"
+         aria-valuemax="50"
+         aria-valuemin="10"
+         aria-valuenow="30"
+         aria-valuetext="30%"
+         class="thumb small notDisabledThumb"
+         data-testid="monday-slider__thumb-1"
+         role="slider"
+         style="left: 50%;"
          tabindex="0"
        />
      </div>
`;
